app/javascript/portal/portalThemeHelper.js

```diff
@@ -0,0 +1,146 @@
+import Cookies from 'js-cookie';
+import { adjustColorForContrast } from '../shared/helpers/colorHelper.js';
+
+const SYSTEM_COOKIE = 'system_theme';
+const SELECTED_COOKIE = 'selected_theme';
+
+export const setPortalHoverColor = theme => {
+  if (window.portalConfig.isPlainLayoutEnabled === 'true') return;
+  const portalColor = window.portalConfig.portalColor;
+  const bgColor = theme === 'dark' ? '#151718' : 'white';
+  const hoverColor = adjustColorForContrast(portalColor, bgColor);
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
+    '--dynamic-hover-color',
+    hoverColor
+  );
+};
+
+export const setPortalClass = theme => {
+  const portalDiv = document.getElementById('portal');
```

Comment:
In multiple methods we do a `document.getElementById(<some-id>);` We can cache this of sorts, let's declare a top level 

`let poratlDiv = document.getElementById('portal');`

And then use this across the script. That way we will ignore doing these queries all the time

Reply:
@scmmishra I tried it, but the class is not adding in the portalDiv element. Any idea?

Reply:
We can add a null check, so 

```
let portalDiv = document.getElementById('portal');

export const setPortalClass = theme => {
    portlalEl = portalDiv ?? document.getElementById('portal');
```

Reply:
https://github.com/chatwoot/chatwoot/pull/8327

Reply:
Done